libs/testserver/server.go

```diff
@@ -52,6 +69,22 @@ func (s *Server) Handle(pattern string, handler HandlerFunc) {
 			return
 		}
 
+		if s.RecordRequests {
+			body, err := io.ReadAll(r.Body)
+			assert.NoError(s.t, err)
+
+			var reqBody map[string]any
+			err = json.Unmarshal(body, &reqBody)
+			assert.NoError(s.t, err)
```

Comment:
why marshal / unmarshal here? Why not just write it as is?

Reply:
Done. typecasting the body to `json.RawMessage` to avoid ending up with a double encoded JSON in `out.request.json`.

Otherwise it'll look like:
```
[
  {
    "method": "POST",
    "path": "/api/2.1/jobs/create",
    "body": "{"name": "abc"}" # with the appropriate string encoding...
  }
]
```